spec/rubocop/cop/lint/symbol_conversion_spec.rb

```diff
@@ -154,6 +158,19 @@
           { foo: :bar }
         RUBY
       end
+
+      it 'registers an offense for dstr' do
+        # rubocop:disable Lint/LiteralInInterpolation
+        expect_offense(<<~RUBY)
+          { foo: "bar-#{'baz'}".to_sym }
+                 ^^^^^^^^^^^^^^^^ Unnecessary symbol conversion; use `:"bar-#{'baz'}"` instead.
+        RUBY
+
+        expect_correction(<<~RUBY)
+          { foo: :"bar-#{'baz'}" }
+        RUBY
+        # rubocop:enable Lint/LiteralInInterpolation
```

Comment:
> The 5 characters from the literal interpolation are discarded and as such the range is off by that many characters. I have no idea where to start with that. I believe that issue may be unrelated to this PR.

The heredoc needs to be equivalent to single quotes. The following should resolve it:

```suggestion
        expect_offense(<<~'RUBY')
          { foo: "bar-#{'baz'}".to_sym }
                 ^^^^^^^^^^^^^^^^^^^^^ Unnecessary symbol conversion; use `:"bar-#{'baz'}"` instead.
        RUBY

        expect_correction(<<~'RUBY')
          { foo: :"bar-#{'baz'}" }
        RUBY
```